### PR TITLE
Separate OCP deployment

### PR DESCRIPTION
**Pull request by petr-balogh**
Signed-off-by: Petr Balogh <pbalogh@redhat.com>

### CODE REVIEW

File: ocs_ci/deployment/ocp.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        logger.info("Destroying the cluster")
+        destroy_cmd = (
+            f"{self.installer} destroy cluster "
+            f"--dir {self.cluster_path} "
+            f"--log-level {log_level}"
+        )
+
+        try:
+
+            # Retrieve cluster name and AWS region from metadata
```
Comment (dahorak) [inline ocs_ci/deployment/ocp.py]: This shouldn't be specific for AWS, right? (probably just the comment mentions it incorrectly)

File: ocs_ci/deployment/deployment.py [hunk 2]
```diff
@@ -34,8 +29,13 @@ def __init__(self):
         self.platform = config.ENV_DATA['platform']
         self.ocp_deployment_type = config.ENV_DATA['deployment_type']
         self.cluster_path = config.ENV_DATA['cluster_path']
-        self.deploy = config.RUN['cli_params']['deploy']
-        self.teardown = config.RUN['cli_params']['teardown']
+
+    class OcpDeployment(BaseOcpDeployment):
```
Comment (vasukulkarni) [inline ocs_ci/deployment/deployment.py]: s/OcpDeployment/OCPDeployment/

File: ocs_ci/deployment/ocp.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.openshift_ops import OCP
+from ocs_ci.utility import utils, templating, system
+from ocs_ci.utility.utils import run_cmd
+
+
+logger = logging.getLogger(__name__)
+
+
+class OcpDeployment:
```
Comment (vasukulkarni) [inline ocs_ci/deployment/ocp.py]: same

**Comment by petr-balogh** [comment]
Testing running here:  https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/344/console

**Comment by petr-balogh** [comment]
Deploy, Teardown passed well - tier1 tests some failures but as I don't change any OCS deployment logic in this PR the failures are not related to this PR.

**Pull request merged**